pkg/appgw/backendaddresspools.go

```diff
@@ -162,6 +162,17 @@ func getAddressesForSubset(subset v1.EndpointSubset) *[]n.ApplicationGatewayBack
 		}
 	}
 
+	for _, address := range subset.NotReadyAddresses {
```

Comment:
Reading the associated issue, I understand why we are also populating App Gateway with the `NotReadyAddresses`.
I worry that the next person to browse through this file may not have the context, and perhaps intuition would be to think - this is not right.

With this PR or later we should add a comment to explain why we have decided to program the NotReadyAddresses as well.